### PR TITLE
[MU4] Midi input port

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: framework/midi/midimodule.cpp [hunk 1]
```diff
@@ -42,20 +43,28 @@ using namespace mu::midi;
...
 #ifdef Q_OS_MACOS
 //#include "internal/platform/osx/coremidioutport.h"
+//#include "internal/platform/osx/coremidiinport.h"
 //static std::shared_ptr<IMidiOutPort> midiOutPort = std::make_shared<CoreMidiOutPort>();
+//static std::shared_ptr<IMidiInPort> midiInPort = std::make_shared<CoreMidiInPort>();
 
 #include "internal/dummymidioutport.h"
+#include "internal/dummymidiinport.h"
 static std::shared_ptr<IMidiOutPort> midiOutPort = std::make_shared<DummyMidiOutPort>();
+static std::shared_ptr<IMidiOutPort> midiOutPort = std::make_shared<DummyMidiInPort>();
```
Comment (psmokotnin) [inline framework/midi/midimodule.cpp:67]: line 67 duplicates 66

Comment (Jojo-Schmitz) [reply framework/midi/midimodule.cpp:67]: Strange that this even passes the compiler

Comment (igorkorsukov) [reply framework/midi/midimodule.cpp:67]: ups 
This is code only for MacOS. 
I didn't compile for MacOS, I asked @psmokotnin  check compilation and fix it

Comment (igorkorsukov) [reply framework/midi/midimodule.cpp]: fixed

File: framework/midi/internal/dummymidioutport.cpp [hunk 2]
```diff
@@ -48,12 +49,16 @@ bool DummyMidiOutPort::isConnected() const
...
 }
 
-void DummyMidiOutPort::sendEvent(const Event& e)
+mu::Ret DummyMidiOutPort::sendEvent(const Event& e)
 {
+    if (!isConnected()) {
+        return make_ret(Err::MidiNotConnected);
+    }
     LOGI() << e.to_string();
+    return make_ret(Err::NoError);
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply framework/midi/internal/dummymidioutport.cpp]: changed on `return Ret(true);`

**Pull request merged**